### PR TITLE
League statistics1.0 

### PR DESCRIPTION
**Pull request by nurudeensulaimon**
## Description

Please include a description of what was changed

## Type of change

- [ ] fix
- [ ] feat
- [ ] test
- [ x] refactor
- [ ] docs

## Checklist

- [ ] code has been self reviewed
- [ ] code runs without any errors
- [ ] thorough testing has been implemented if adding feature
- [ ] all tests pass

Refactored worst_offense_team method and the data issue is fixed locally.

### CODE REVIEW

File: lib/league_statistics.rb [hunk 1]
```diff
@@ -1,51 +1,81 @@
...
+        games = @game_team_data.find_all do |game_team|
+        game_team[:team_id] == team_id    
+   end 
+        total_goals = games.map do |game|
+        game[:goals].to_i 
+   end.sum
+        average_goals_per_season = total_goals / games.length  
+  end  
+end 
+
```
Comment (kbug819) [inline lib/league_statistics.rb:116]: Great work on this method!

**Comment by kbug819** [comment]
Awesome progress, don't forget to continue committing periodically as you work on your next methods!

**Comment by elizakeating** [comment]
Nice helper method! Super useful!

File: lib/league_statistics.rb [hunk 2]
```diff
@@ -1,51 +1,162 @@
...
+      def lowest_scoring_hometeam
+
+        team_ids = @game_team_data[:team_id].uniq
+        highest_scoring_hometeam = team_ids.min_by do |team_id|
+          average_goals_by_hometeam(team_id)
+     end  
+         team = @teams_data.find do |team|
+         team[:team_id] == highest_scoring_hometeam
+    end 
+          team[:teamname]
```
Comment (kbug819) [inline lib/league_statistics.rb]: Let's work a bit on spacing

File: lib/league_statistics.rb [hunk 3]
```diff
@@ -1,51 +1,162 @@
...
-    games = read_csv(@file)
+    def  average_goals_by_visitor(team_id)
+  
+            away_games = @game_team_data.find_all do |game_team|
+                       game_team[:hoa]== "away"
+                       game_team[:team_id] == team_id 
+        end
+             total_away_goals = away_games.map do |away_game|
+                      away_game[:goals].to_i 
+        end.sum
```
Comment (kbug819) [inline lib/league_statistics.rb]: Great work on these helper methods for averages for visitors and home teams.

**Comment by elizakeating** [comment]
Woo! Nice methods and testing! Thanks for cleaning up the spacing of your methods!

**Comment by kbug819** [comment]
Great work!

**Review by ethanb1145 — COMMENTED** [review]
Nice formatting, code looks fixed and organized now. Good job with completing the last methods for this class and making them pass the tests.

**Pull request merged**